File: Python/동전 0.py
N,K = map(int, input().split())

money = list()
cnt = 0

for i in range(0,N):
    money.append(int(input()))

for i in range(N-1,-1,-1):
    if K >= money[i]:
        cnt += K // money[i]
        K %= money[i]
    elif K == 0:
        break
print(cnt)

File: JAVA/Greedy/src/동전0.java
import java.util.*;
public class 동전0 {
    public static void main(String[] args)
    {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        int K = scanner.nextInt();
        Integer[] money = new Integer[N];
        for(int i=0; i<N; i++)
        {
            money[i] = scanner.nextInt();
        }
        Arrays.sort(money, Collections.reverseOrder());

        int cnt = 0;
        for(int i=0; i<N; i++)
        {
            if(K>=money[i])
            {
                cnt += K/money[i];
                K %= money[i];
            }
            else if(K==0)
                break;
        }
        System.out.print(cnt);
    }
}